driver/razerchromacommon.c

```diff
@@ -1546,3 +1546,77 @@ struct razer_report razer_chroma_misc_get_scroll_smart_reel(void)
 
     return report;
 }
+
+/**
+ * Set LED mode for HyperPolling Wireless Dongle
+ * 1 = Connection Status
+ * 2 = Battery Status
+ * 3 = Battery Warning Only
+ */
+struct razer_report razer_chroma_misc_set_hyperpolling_wireless_dongle_indicator_led_mode(unsigned char mode)
+{
+    struct razer_report report = get_razer_report(0x07, 0x10, 0x01);
+
+    if(mode < 0x01 || mode > 0x03) {
+        mode = 0x01;
+    }
+
+    report.arguments[0] = mode;
+
+    return report;
+}
+
+/**
+ * Get LED mode for HyperPolling Wireless Dongle
+ */
+struct razer_report razer_chroma_misc_get_hyperpolling_wireless_dongle_indicator_led_mode(void)
+{
+    struct razer_report report = get_razer_report(0x07, 0x90, 0x01);
+
+    return report;
+}
+
+/**
+ * Set pairing mode for HyperPolling Wireless Dongle
+ */
+struct razer_report razer_chroma_misc_set_hyperpolling_wireless_dongle_pair(unsigned int pid, int step)
+{
+    struct razer_report report;
+
+    if (step == 1) {  // Put dongle into pairing mode
+        report = get_razer_report(0x00, 0x46, 0x01);
+
+        report.arguments[0] = 0x01;
+    }
+
+    else if (step == 2) {  // Pair with PID
+        report = get_razer_report(0x00, 0x41, 0x03);
+
+        if(pid < 0x01 || pid > 0xff) {
+            pid = 0x00;  // Set an invalid PID (harmless)
+        }
+
+        report.arguments[0] = 0x01;
+        report.arguments[1] = 0x00;
+        report.arguments[2] = pid;
+    }
+
+    return report;
```

Comment:
Please just make two different methods out of this to avoid this special logic.

Reply:
Did you mean to split the 2 parts of the pairing process into individual methods? I combined them because they always go together according to the pcaps, there wouldn't be a need to perform one part separately from the other. Nevertheless, I split the pair method into 2 different methods :)